dll/win32/advapi32/service/sctrl.c

```diff
@@ -991,25 +1003,24 @@ StartServiceCtrlDispatcherA(const SERVICE_TABLE_ENTRYA *lpServiceStartTable)
         i++;
     }
 
-    dwActiveServiceCount = i;
-
     /* Allocate the service table */
     lpActiveServices = RtlAllocateHeap(RtlGetProcessHeap(),
                                        HEAP_ZERO_MEMORY,
-                                       dwActiveServiceCount * sizeof(ACTIVE_SERVICE));
+                                       i * sizeof(ACTIVE_SERVICE));
     if (lpActiveServices == NULL)
```

Comment:
it is not obvious to me what prevents a later NULL-deref of `lpActiveServices`, as `dwActiveServiceCount` remains set.